### ISSUE TITLE
Add conversion from ethology dataset to torch dataset / dataloader

### ISSUE THREAD

**Issue by sfmig** [neuroinformatics-unit/ethology#51]
**Is your feature request related to a problem? Please describe.**
A module to define a Pytorch dataset & dataloader from a set of annotations + images. For now, we can focus on the detection case.

**Describe the solution you'd like**
Taking as input an annotations dataframe and an image directory, add functionality to support and review a labelled dataset.

**Describe alternatives you've considered**
\

**Additional context**
\

**Comment by hrigar** [comment]
hi, i want to work on this , i will soon crate a draft PR, let me know if that works.

**Comment by hrigar** [comment]
I've implemented the basic dataset module that loads images and their annotations. For the detection part, we have two options:

1. Current Basic Implementation:
-  Loads images from directory
-  Associates with annotation dataframe


2. Should we extend to detection now?
We could add:
- Bounding box handling
- Label processing
- Box format conversions

just want to make sure i am going in the right direction.
Let me know what do u think, open to changes and feedback.

**Comment by sfmig** [comment]
this xbatcher library looks great: https://github.com/xarray-contrib/xbatcher
